From 0f334f0407859331f715bb78616e10319d7f925d Mon Sep 17 00:00:00 2001
From: alheio <zaharov.sy@gmail.com>
Date: Fri, 12 Dec 2014 16:57:21 +0300
Subject: [PATCH 0154/1173] [oozie] Order coordinator actions on the Oozie
 side

https://github.com/cloudera/hue/pull/132

Oozie configuration can have 'oozie.coord.actions.default.length' parameter which overloads oozie 'len'
param, and Hue may not display latest coordinator actions if they are exceeding this value.
Since https://issues.apache.org/jira/browse/OOZIE-1754 oozie can sort coordinator actions.
We need to request oozie to sort coordinator actions in descending order and display them
correctly by dashboard.py.
---
 apps/oozie/src/oozie/views/dashboard.py         |    5 ++++-
 desktop/libs/liboozie/src/liboozie/oozie_api.py |    1 +
 2 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/apps/oozie/src/oozie/views/dashboard.py b/apps/oozie/src/oozie/views/dashboard.py
index 51a5d07..4542dc3 100644
--- a/apps/oozie/src/oozie/views/dashboard.py
+++ b/apps/oozie/src/oozie/views/dashboard.py
@@ -740,7 +740,10 @@ def massaged_coordinator_actions_for_json(coordinator, oozie_bundle):
       'missingDependencies': action.missingDependencies
     }
 
-    actions.insert(0, massaged_action)
+    actions.append(massaged_action)
+
+  # Sorting for Oozie < 4.1 backward compatibility
+  actions.sort(key=lambda k: k['number'], reverse=True)
 
   return actions
 
diff --git a/desktop/libs/liboozie/src/liboozie/oozie_api.py b/desktop/libs/liboozie/src/liboozie/oozie_api.py
index ab01ce6..4ce2229 100644
--- a/desktop/libs/liboozie/src/liboozie/oozie_api.py
+++ b/desktop/libs/liboozie/src/liboozie/oozie_api.py
@@ -160,6 +160,7 @@ class OozieApi(object):
   def get_coordinator(self, jobid):
     params = self._get_params()
     params.update({'len': -1})
+    params.update({'order': 'desc'})
     resp = self._root.get('job/%s' % (jobid,), params)
     return Coordinator(self, resp)
 
-- 
1.7.9.5

